Oryx.Cognite/src/Beta/Handler.fs

```diff
@@ -21,64 +19,44 @@ open CogniteSdk
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 [<AutoOpen>]
 module Handler =
-    let withBeta (next: NextFunc<_,_>) (context: HttpContext) =
-        withVersion
-            V10
-            next
-            { context with Request = { context.Request with Headers = context.Request.Headers.Add("version", "beta") } }
+    let withBetaHeader next ctx =
+        withHeader "version" "beta" next ctx
 
-    let getBeta<'a, 'b> (url: string)  : HttpHandler<HttpResponseMessage, 'a, 'b> =
-        withBeta  >=> get url
+    let withBetaVersion next ctx =
```

Comment:
If you apply some lambda calculus here you should end up with:
```fs
let withBetaVersion =
    withBetaHeader
    >=> withVersion V10
```

Reply:
https://wiki.haskell.org/Eta_conversion

Reply:
Yes, but then it won't compile because all succeeding functions aren't sufficiently generic. 😂

Reply:
Strange, let's figure it out later ....